util/src/kvdb.rs

```diff
@@ -208,8 +208,13 @@ pub struct Database {
 	config: DatabaseConfig,
 	write_opts: WriteOptions,
 	read_opts: ReadOptions,
-	overlay: RwLock<Vec<HashMap<ElasticArray32<u8>, KeyState>>>,
 	path: String,
+	// Dirty values added with `write_buffered`. Cleaned on `flush`.
+	overlay: RwLock<Vec<HashMap<ElasticArray32<u8>, KeyState>>>,
+	// Values currently being flushed. Cleared when `flush` completes.
+	flushing: RwLock<Vec<HashMap<ElasticArray32<u8>, KeyState>>>,
```

Comment:
An optimization introduced here to allow DB queries while `flush` is in progress